core/includes/install.inc

```diff
@@ -705,17 +705,20 @@ function backdrop_install_config_directories() {
     // public files directory, which has already been verified to be writable
     // itself. But if it somehow fails anyway, the installation cannot proceed.
     // Bail out using a similar error message as in system_requirements().
-    $config_directory = config_get_config_directory($config_type);
-    if (!file_prepare_directory($config_directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS)) {
-      throw new Exception(format_string('The directory %directory could not be created or could not be made writable. To proceed with the installation, either create the directory and modify its permissions manually or ensure that the installer has the permissions to create it automatically. For more information, see the <a href="@handbook_url">Installation Instructions</a> page.', array(
-        '%directory' => $config_directory,
-        '@handbook_url' => 'https://backdropcms.org/installation',
-      )));
+   $config_storage = config_get_config_storage($config_type);
+    try {
+      $config_storage->initializeStorage();
+    }
+    catch (\ConfigStorageException $e) {
+      throw new Exception(format_string('@storage_error.  To proceed with the installation, either create the directory and modify its permissions manually or ensure that the installer has the permissions to create it automatically. For more information, see the <a href="@handbook_url">Installation Instructions</a> page.', array(
+          '@storage_error' => $e->getMessage(),
+          '@handbook_url' => 'https://backdropcms.org/installation',
+        )), 0, $e);
```

Comment:
:+1: Nice job on making this a new method in the config storage class instead of having storage code directly in this function.